syntax = "proto3";

package inventory;

service InventoryService {
  rpc CheckStock (StockRequest) returns (StockResponse);
  rpc ReserveStock (ReserveRequest) returns (ReserveResponse);
  rpc ConfirmStockReduction (ConfirmRequest) returns (ConfirmResponse);
}

message StockRequest {
  string book_id = 1;
}

message StockResponse {
  string book_id = 1;
  int32 available_units = 2;
  bool in_stock = 3;
}

message ReserveRequest {
  string book_id = 1;
  int32 quantity = 2;
}

message ReserveResponse {
  bool success = 1;
  string message = 2;
}

message ConfirmRequest {
  string book_id = 1;
  int32 quantity = 2;
}

message ConfirmResponse {
  bool confirmed = 1;
  string message = 2;
}
